### PR TITLE
Use HTTPS for instances that support it

### LINKED ISSUE #4: Point to HTTPS onion address for onion services that support it
So far all the .onion addresses I've tested with did not use an EV SSL cert; it'd be good to document the behavior for .onion addresses that use one (e.g., is the EV information still shown?).

### PR DESCRIPTION
**Pull request by legoktm**
### Status

Ready for review; needs prod signature

### Description

For instances that already redirect to HTTPS, have our ruleset also
point to HTTPS, avoiding one extra HTTP->HTTPS redirect.
    
Since this information isn't available from the directory, add it as an
optional field to onboarded.txt, which shouldn't be too much extra
maintenance given the few instances that use it.
    
Fixes #4.

### Review Checklist

- [x] Changes to `onboarded.txt` are accurate
- [x] The file `default.rulesets.TIMESTAMP.gz` has been updated, extracting that file and inspecting the contents of the JSON file produces the expected rules
- [x] The ruleset has been verified by modifying the HTTPS Everywhere configuration in a Tor Browser instance pointing to `Path Prefix`: `https://raw.githubusercontent.com/freedomofpress/securedrop-https-everywhere-ruleset/$BRANCH_NAME`
- [x] `index.html` has been updated using `./update_index.sh`

### Post-Deployment Checklist

- [ ] Added/modified onion names have been updated in the SecureDrop Directory

### CODE REVIEW

**Comment by legoktm** [comment]
A far simpler approach would be to add a `https` column to `onboarded.txt` and use that. I initially dismissed that but given there are just 2 instances that need it, it seems pretty straightforward to maintain.

**Review by zenmonkeykstop — CHANGES_REQUESTED** [review]

File: sddir.py [hunk 1]
```diff
@@ -3,17 +3,22 @@
...
 from typing import Dict
 
 # Configure logging output
 logfmt = "%(asctime)s %(levelname)-8s %(message)s"
 logging.basicConfig(format=logfmt, level=logging.DEBUG, datefmt="%Y-%m-%d %H:%M:%S")
 
 SECUREDROP_ONION_PSEUDO_TLD = ".securedrop.tor.onion"
-DEFAULT_ONION_PROTOCOL = "http://"  # We don't store protocol in the directory
 RULESET_DIR = "rulesets"
+# Use Tor Browser's internal SOCKS proxy
```
[the remark this answers is not in the document]
Comment (legoktm) [reply sddir.py]: That was my intention, but no longer in the `onboarded.txt` version.

**Comment by legoktm** [comment]
Now updated to store and pull data from onboarded.txt instead of doing live checking.

**Comment by zenmonkeykstop** [comment]
Confirmed that the ruleset changes look consistent with the https option. A new signed ruleset has been added in a separate commit. Final review to follow to verify the rules work in TBB as expected.

**Comment by zenmonkeykstop** [comment]
(handing off to @nathandyer for live ruleset checks)

**Review by nathandyer — APPROVED** [review]
Happy to report everything worked correctly for me in the live tests, and that the Source Interfaces loaded using HTTPS.

Just a note about the test plan, no changes should be necessary in the directory itself.

**Pull request merged**